lib/cstruct.ml

```diff
@@ -88,10 +88,15 @@ let of_bigarray ?(off=0) ?len buffer =
 let to_bigarray buffer =
   Bigarray.Array1.sub buffer.buffer buffer.off buffer.len
 
-let create len =
+let create_unsafe len =
   let buffer = Bigarray.(Array1.create char c_layout len) in
   { buffer ; len ; off = 0 }
 
+let create len =
+  let t = create_unsafe len in
+  Bigarray.Array1.fill t.buffer (Char.chr 0);
```

Comment:
Is there any benefit? I guess `memset` is only needed if you want to fill a sub-region of an array (`fill` can only fill the whole thing, but that's what we want here).